src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-        logger.info(f"Matched plants:    {len(connected_plant_ids)}")
-        logger.info(f"Not connected:       {len(not_in_ferc1_plant_ids)}")
-        logger.info(f"Missing connections: {len(missing_plant_ids)}")
-        # Investigation of 3 missing connections [1204, 1147, 1223] determined
-        # that these plants are missing from ferc because there was no
-        # reporting for the specific years in question
-        if len(missing_plant_ids) > 3:
-            warnings.warn(
-                f'There are {len(missing_plant_ids)} missing plant records.')
-
-    def get_candidate_matches(self):
-        """
-        Prepare all candidate matches between depreciation and ferc1 steam.
+    Args:
+        treatments: a dictionary of column name (keys) with field treatments
+            (values)
+        eia_pk:
+        data_set_idx_cols:
+        plant_part:
 
-        Before choosing the specific match between depreciation and ferc1, we
-        need to compile all possible options - or candidate links.
+    """
 
-        Returns:
-            pandas.DataFrame: a dataframe with all of the possible combinations
-            of the deprecation data and ferc1 (with their respective EIA master
-            unit list records associated).
-        """
-        # add in a dtype enforcer bc OMIGOSH they keep converting to non-nullables
-        candidate_matches_all = pd.merge(
-            self.inputs.connects_deprish_eia.pipe(
-                pudl.helpers.convert_cols_dtypes, 'ferc1'),
-            self.inputs.connects_ferc1_eia.pipe(
-                pudl.helpers.convert_cols_dtypes, 'ferc1'),
-            on=CONNECT_COLS,
-            suffixes=('_deprish', '_ferc1'),
-            how='left', indicator=True
-        )
-        # reorder cols so they are easier to see, maybe remove later
-        first_cols = ['plant_part_deprish', 'plant_part_ferc1',
-                      'record_id_eia_deprish', 'record_id_eia_ferc1',
-                      'plant_part_name', 'plant_name_match',
-                      'fraction_owned_deprish', 'fraction_owned_ferc1',
-                      'record_count_deprish', 'record_count_ferc1'
-                      ]
-        candidate_matches_all = candidate_matches_all[
-            first_cols + [x for x in candidate_matches_all.columns
-                          if x not in first_cols]]
-
-        self.check_all_candidate_matches(candidate_matches_all)
-
-        # rename dict with the ordered plant part names with numbered prefixes
-        replace_dict = {
-            x:
-            f"{pudl.analysis.plant_parts_eia.PLANT_PARTS_ORDERED.index(x)}_{x}"
-            for x in pudl.analysis.plant_parts_eia.PLANT_PARTS_ORDERED
+    treatments: Dict[str, FieldTreatment]
+    eia_pk: List[str] = ['record_id_eia']
+    data_set_idx_cols: List[str]
+    plant_part: Literal['plant_gen']
+
+    def get_treatment_cols(self, treatment_type: str) -> list[str]:
```

Comment:
Should this be a private method? Does it have an application outside of the properties defined below?

Reply:
good point!

Reply:
hm. well there are two instances of grabbing the list of scale and str_concat columns. Do you think I should just make @propoerty methods for those things as well?